core/src/main/java/org/elasticsearch/index/query/SpanNotQueryBuilder.java

```diff
@@ -19,83 +19,174 @@
-        this.exclude = exclude;
-        return this;
+    // only used for prototype
+    private SpanNotQueryBuilder() {
+           this.include = null;
+           this.exclude = null;
+    }
+
+    /**
+     * @return the span query whose matches are filtered
+     */
+    public SpanQueryBuilder include() {
+        return this.include;
     }
 
+    /**
+     * @return the span query whose matches must not overlap
+     */
+    public SpanQueryBuilder exclude() {
+        return this.exclude;
+    }
+
+    /**
+     * @param dist the amount of tokens from within the include span can’t have overlap with the exclude span.
+     * Equivalent to setting both pre and post parameter.
+     */
     public SpanNotQueryBuilder dist(int dist) {
-        this.dist = dist;
+        pre(dist);
+        post(dist);
         return this;
     }
 
+    /**
+     * @param pre the amount of tokens before the include span that can’t have overlap with the exclude span. Values
+     * smaller than 0 will be ignored and 0 used instead.
+     */
     public SpanNotQueryBuilder pre(int pre) {
-        this.pre = (pre >=0) ? pre : 0;
+        this.pre = (pre >= 0) ? pre : 0;
         return this;
     }
 
+    /**
+     * @return the amount of tokens before the include span that can’t have overlap with the exclude span.
+     * @see SpanNotQueryBuilder#pre(int)
+     */
+    public Integer pre() {
+        return this.pre;
+    }
+
+    /**
+     * @param post the amount of tokens after the include span that can’t have overlap with the exclude span.
+     */
     public SpanNotQueryBuilder post(int post) {
         this.post = (post >= 0) ? post : 0;
         return this;
     }
 
+    /**
+     * @return the amount of tokens after the include span that can’t have overlap with the exclude span.
+     * @see SpanNotQueryBuilder#post(int)
+     */
+    public Integer post() {
+        return this.post;
+    }
+
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
-        if (include == null) {
-            throw new IllegalArgumentException("Must specify include when using spanNot query");
-        }
-        if (exclude == null) {
-            throw new IllegalArgumentException("Must specify exclude when using spanNot query");
-        }
-
-        if (dist != null && (pre != null || post != null)) {
-             throw new IllegalArgumentException("spanNot can either use [dist] or [pre] & [post] (or none)");
-        }
-
         builder.startObject(NAME);
         builder.field("include");
         include.toXContent(builder, params);
         builder.field("exclude");
         exclude.toXContent(builder, params);
-        if (dist != null) {
-            builder.field("dist", dist);
-        }
-        if (pre != null) {
-            builder.field("pre", pre);
-        }
-        if (post != null) {
-            builder.field("post", post);
-        }
```

Comment:
Setting dist is equivalent to setting pre/post to the same value, so I removed the internal representation for dist entirely. Only caveat here is that if user sets dist on the builder, query is logged with per/post fields, but since we said doXContent will be used for logging / debugging only after refactoring this is similar to outputting defaults that user hasn't explicitely set, I think.